File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    valid = []
    for lotto in lottos:
        if lotto == 0:
            valid.append(-1)
        elif lotto in win_nums:
            valid.append(1)
        else:
            valid.append(0)
    cnt_zero = valid.count(-1)
    total = sum(valid)
    if total == 0 and cnt_zero == 0:
        return [6, 6]
    else:
        return [(7-(total+2*cnt_zero)), (7-(total+cnt_zero))] if cnt_zero < 6 else [1, 6]
    
    
    

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int count = 0;
        int zero = 0;
        List<Integer> list = new ArrayList<>();
        for (int lotto : lottos) {
            if (lotto == 0)
                ++zero;
            else
                list.add(lotto);
        }
        for (int num : win_nums) {
            if (list.contains(num))
                ++count;
        }
        int[] answer = new int[2];
        answer[0] = 7-count-zero;
        answer[1] = 7-count;
        if (answer[0] == 7)
            answer[0] = 6;
        if (answer[1] == 7)
            answer[1] = 6;
        
        return answer;
    }
}